module/x/gravity/client/cli/query.go

```diff
@@ -830,6 +838,239 @@ func CmdLastObservedEthereumHeight() *cobra.Command {
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.BatchTxConfirmationsByValidator(cmd.Context(), &types.BatchTxConfirmationsByValidatorRequest{
+				ValidatorAddress: val.String(),
+			})
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+func CmdContractCallTxConfirmationsByValidator() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "contract-call-tx-ethereum-signatures-by-validator [validator-address]",
+		Args:  cobra.ExactArgs(1),
+		Short: "query contract call tx signatures submitted by a given validator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, queryClient, err := newContextAndQueryClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			val, err := sdk.ValAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.ContractCallTxConfirmationsByValidator(cmd.Context(), &types.ContractCallTxConfirmationsByValidatorRequest{
+				ValidatorAddress: val.String(),
+			})
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+func CmdSignerSetTxConfirmationsByValidator() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "signer-set-tx-ethereum-signatures-by-validator [validator-address]",
+		Args:  cobra.ExactArgs(1),
+		Short: "query signer set tx signatures submitted by a given validator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, queryClient, err := newContextAndQueryClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			val, err := sdk.ValAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.SignerSetTxConfirmationsByValidator(cmd.Context(), &types.SignerSetTxConfirmationsByValidatorRequest{
+				ValidatorAddress: val.String(),
+			})
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+
+func CmdEthereumEventVoteRecords() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ethereum-event-vote-records",
+		Args:  cobra.NoArgs,
+		Short: "query ethereum event vote records that have not been pruned",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, queryClient, err := newContextAndQueryClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			res, err := queryClient.EthereumEventVoteRecords(cmd.Context(), &types.EthereumEventVoteRecordsRequest{
+				Pagination: pageReq,
+			})
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	flags.AddPaginationFlagsToCmd(cmd, "all ethereum event vote records")
+	return cmd
+}
+
+func CmdEthereumEventVotes() *cobra.Command {
```

Comment:
Is there a functional difference between "Event vote records" in the previous command and "Event votes" here? Is the difference just specifying a validator address? If so we should refer to them as the same thing.

Reply:
A record contains all votes, a vote is associated with an address. For the CLI I could just say "vote" and "votes" to be more consistent with other commands.